core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala

```diff
@@ -494,6 +500,20 @@ abstract class Stream[+T] { self =>
     ret
   }
 
+  /**
+    * Returns a continuous even that is true when condition is true
+    * The pollingSource ImpulseEvent is used to check when condition is true
+    * @param condition condition of the event
+    * @param pollingSource source of when to update the check for condition
+    * @return
+    */
+  def evenWithCondition(condition: T => Boolean)
+                       (implicit pollingSource: ImpulseEvent): ContinuousEvent = {
+    new ContinuousEvent(
```

Comment:
I think we can completely eliminate pollingSources at this point and update the continuous event on each stream update.